sdk/keyvault/pom.service.xml

```diff
@@ -36,10 +36,12 @@
         </property>
       </activation>
       <modules>
-        <module>../core/azure-core</module>        <!-- NOTE: We need to build core so we can build against the version in the repo. -->
-        <module>../core/azure-core-test</module>        <!-- NOTE: We need to build core so we can build against the version in the repo. -->
+      <!-- NOTE: Commenting out because we should now be binary composed.
+        -->
+        <module>../core/azure-core</module>      
```

Comment:
Sadly no ... tried and it caused a bunch of issues. It's because the packages that we've published aren't the LTS ones with the module-info file. It meant that the LTS builds against the packages failed. it also caused some related issues in Java 8 builds too.

So the next time we service we should look at releasing the LTS versions of the packages.

Reply:
Yeah, LTS builds will fail :( but curious to know what were the issues with Java 8 builds.

@JonathanGiles What is your recommendation on which jars should be published to Maven? Should it be LTS or JDK 8 or both?

Reply:
It was complaining about the finding some types in com.azure:azure-core (via the @link annotations). They were JavaDoc warnings which we of course treat as failures.

https://dev.azure.com/azure-sdk/public/_build/results?buildId=174366&view=logs&j=9d510ff3-53a8-5192-bd2d-466ce830bc13&t=01de8c6b-9f4b-5017-894e-951524cf562e&l=17

Its easy enough to reproduce locally. Just comment out the core modules in pom.service.xml and then run the package goal against pom.service.xml.

Reply:
My PR at #6080 does away with separate 8 and LTS builds. In that world, we would be releasing the LTS variant, but compiled as a release targeting JDK 8.